server/routes/user.js

```diff
@@ -0,0 +1,37 @@
+const express = require('express');
+const router = express.Router();
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const User = require('../database/models/user');
+const registrationValidator = require('../controllers/userRegistrationValidator');
+const loginValidator = require('../controllers/userLoginValidator');
+
+router.post('/register', (req, res) => {
+  const {validationErrors, isRegistrationValid} = registrationValidator(req.body);
+
+  if(!isRegistrationValid) {
+    return res.status(400).json(validationErrors);
+  }
+
+  User.findOne({ email: req.body.email })
+      .then(user => {
+        if(user) {
+          return res.status(400).json({email: "Email you entered already exists"});
+        }
+        else {
+          const newUser = new User({
+            email: req.body.email,
+            password: req.body.password
+          });
+          bcrypt.genSalt(12, (err, salt) => {
+            bcrypt.hash(newUser.password, salt, (err, hash) => {
+              if(err) throw err;
+              newUser.password = hash;
+              newUser.save().then(user => res.json(user))
+                            .catch(err => console.error(err));
```

Comment:
in mongoose docs, I see a lot of this: `handleError()`. is this a helper defined inside of mongoose or is it something I need to define? I was not clear if this is preferred to console.error()

Reply:
There is nothing special about handleError. Basically every mongoose method provides a way to handle errors. SO its up to you how you define it. I have mostly preferred to use catch block and handle error there. Here is a similar question on stackoverflow:
https://stackoverflow.com/questions/12066426/mongoose-error-handling-in-one-place